lib/pbench/server/api/resources/users_api.py

```diff
@@ -188,6 +194,25 @@ def post(self):
             self.logger.warning("Password not provided during the login process")
             abort(HTTPStatus.BAD_REQUEST, message="Please provide a valid password")
 
+        token_expiry = post_data.get("token_expiry")
+        if token_expiry:
+            if len(token_expiry.keys()) != 1:
+                self.logger.warning(f"Invalid token expiry format {token_expiry}")
+                abort(
+                    HTTPStatus.BAD_REQUEST,
+                    message="Please provide a valid token expiry",
+                )
+            else:
+                if token_expiry.keys() not in self.TOKEN_EXPIRY_KEYS:
+                    self.logger.warning(f"Invalid token expiry format {token_expiry}")
+                    abort(
+                        HTTPStatus.BAD_REQUEST,
+                        message=f"Please provide a valid token expiry, accepted keys are: {self.TOKEN_EXPIRY_KEYS}",
+                    )
+            token_expiry = int(token_expiry)
+        else:
+            token_expiry = {"minutes": int(self.token_expire_duration)}
```

Comment:
is `next(iter(token_expiry))` gonna return all the keys of the `dict` if not used inside the loop, I tried to run it but it only returns the first key of the dict.

Reply:
In the second `elif` of my alternative code, there is exactly one key in `token_expiry`; otherwise, control would have entered the block in the first `elif` instead.

`next()` simply returns the "next" item in its argument; if you reconstruct the argument every time (as the code does here), it always returns the first item.  If you want to move past the first, then you need to construct the iterator, assign it to a variable, and then have `next()` reference the variable.

The use of `next(iter(token_expiry))` is just a means for accessing the first key in `token_expiry`.  I could have used `list(token_expiry.keys())[0]`, but that's much harder to read.  The tricky part is that we know that there is a key, but we don't know what it is, so we need to iterate over the keys one way or another, and using `next()` is handy.